hangups/conversation.py

```diff
@@ -805,43 +805,77 @@ def _add_conversation(self, conversation, events=[]):
 
     @asyncio.coroutine
     def _on_state_update(self, state_update):
-        """Receive a StateUpdate and fan out to Conversations."""
+        """Receive a StateUpdate and fan out to Conversations.
+
+        Args:
+            state_update: hangouts_pb2.StateUpdate instance
+        """
         # Handle updating a conversation
         if state_update.HasField('conversation'):
             self._handle_conversation(state_update.conversation)
+
         # Handle the notification
         notification_type = state_update.WhichOneof('state_update')
+        if notification_type not in ('typing_notification',
+                                     'watermark_notification',
+                                     'event_notification'):
+            return
+
+        conv_id = (state_update.event_notification.event.conversation_id.id
+                   if notification_type == 'event_notification' else
+                   getattr(state_update, notification_type).conversation_id.id)
```

Comment:
Do you think it would be cleaner to move this to the individual handler methods, so this method doesn't have to know to extract the conversation ID from every type of state update?